packages/vite/src/node/ssr/ssrModuleLoader.ts

```diff
@@ -38,7 +38,7 @@ export async function ssrLoadModule(
   urlStack: string[] = [],
   fixStacktrace?: boolean
 ): Promise<SSRModule> {
-  url = unwrapId(url).replace(NULL_BYTE_PLACEHOLDER, '\0')
+  url = unwrapId(url)
```

Comment:
I think the `\0` replacement is needed. If we do an `unwrapId`, we are trying to go from a browser-friendly URL to a valid rollup URL (what plugins expect). I think this is part of not having a helper like `toRollupURL` that does all the needed transforms. 
With this line at the start of `ssrLoadModule`, it means it can receive both a browser-friendly URL or a Rollup URL and it will normalize it. I think we may want to do this with `transformRequest` too later, as I said in my prev comment